demo.cc

```diff
@@ -72,6 +74,28 @@ struct Box {
   std::map<schema::QID, std::map<schema::UID, schema::ANSWER>> answers;
 };
 
+struct TimeWindowTracker {
+  const double w_;
+  std::queue<double> q_;
+  std::mutex mutex_;  // TODO(dkorolev): Remove it.
+  explicit TimeWindowTracker(const double w = 15000.0) : w_(w) {}
+  void Add(double x) {
+    std::lock_guard<std::mutex> guard(mutex_);
+    q_.push(x);
+    Relax(x);
+  }
+  int Get(double x) {
```

Comment:
I don't like the name `Get` - it looks like complement method to `Add` and I expect it to return smth like q.front() :)
Maybe `GetSize` or `GetCount` or just `Size`?